backend/src/lib/report-generator.js

```diff
@@ -305,6 +305,7 @@ function cvssStrToObject(cvss) {
 
 async function prepAuditData(data) {
     var result = {}
+    data = checkIfTemplate(data)
```

Comment:
It would be better to place this at the end of the function with the `result` variable as it will benefit from the transformed data (easier access to custom data like `sections`)